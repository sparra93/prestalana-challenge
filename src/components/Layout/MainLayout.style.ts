import styled from 'styled-components';

export const LayoutContainer = styled.div`
  display: flex;
  flex-direction: column;
  background: rgb(81, 121, 213);
  background: radial-gradient(
    circle,
    rgba(81, 121, 213, 1) 11%,
    rgba(81, 121, 213, 1) 100%,
    rgba(2, 0, 36, 0.8799894957983193) 100%
  );
`;

export const MainContainer = styled.div`
  display: flex;
  flex: 1;
`;

export const Content = styled.div`
  flex: 1;
`;
export const ChildrenContent = styled.div`
  flex: 1;
  padding: 20px;
`;
