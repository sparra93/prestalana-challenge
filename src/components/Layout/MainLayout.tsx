import type { ReactNode } from 'react';
import React from 'react';

import {
  Content,
  LayoutContainer,
  MainContainer,
} from '@/components/Layout/MainLayout.style';
import Sidebar from '@/components/Sidebar';
import Toolbar from '@/components/Toolbar';

interface MainLayoutProps {
  children: ReactNode;
}

const MainLayout: React.FC<MainLayoutProps> = ({ children }) => {
  return (
    <LayoutContainer>
      <Toolbar />
      <MainContainer>
        <Sidebar />
        <Content>{children}</Content>
      </MainContainer>
    </LayoutContainer>
  );
};

export default MainLayout;
