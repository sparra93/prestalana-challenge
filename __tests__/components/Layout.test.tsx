import { screen } from '@testing-library/react';
import { useRouter } from 'next/router';
import React from 'react';

import Layout from '@/components/Layout/MainLayout';

jest.mock('next/router', () => ({
  useRouter: jest.fn(),
}));

describe('Layout Component', () => {
  beforeAll(() => {
    (useRouter as jest.Mock).mockImplementation(() => ({
      pathname: '/',
      push: jest.fn(),
    }));
  });

  it('should display layout tree correctly', () => {
    renderWithProviders(
      <Layout>
        <div>Content</div>
      </Layout>,
    );
    expect(screen.getByText('Content')).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { asFragment } = renderWithProviders(
      <Layout>
        <div>Content</div>
      </Layout>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
