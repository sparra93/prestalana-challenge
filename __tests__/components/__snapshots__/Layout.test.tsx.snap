// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-d0ae5099-0 kwXVNk"
  >
    <header
      class="sc-ece30d93-0 hPavrx"
    >
      <h1
        class="sc-ece30d93-1 dQbGoR"
      >
        My Personal Dashboard
      </h1>
      <a
        data-testid="go-to-cart"
        href="/Cart"
      >
        <div
          class="sc-ece30d93-2 ejPOtw"
        >
          <img
            alt="Cart Icon"
            class="sc-ece30d93-3 cwysBr"
            src="/assets/icons/shopping-bag.png"
          />
          <div
            class="sc-ece30d93-4 ekOwio"
          >
            0
          </div>
        </div>
      </a>
    </header>
    <div
      class="sc-d0ae5099-1 fOSyUL"
    >
      <section
        class="sc-6d8e35a2-0 bpKibS"
      >
        <div
          class="sc-2b2a1bd8-0 doDeWs"
        >
          <img
            alt="User Avatar"
            class="sc-2b2a1bd8-1 cVpnb"
          />
          <div
            class="sc-2b2a1bd8-2 eVjdKf"
          >
            <span
              class="sc-2b2a1bd8-3 dbyjgu"
            />
            <span
              class="sc-2b2a1bd8-4 BLLUx"
            />
            <button
              class="sc-2b2a1bd8-5 kfcqpw"
            >
              Cerrar Sesi√≥n
            </button>
          </div>
        </div>
      </section>
      <div
        class="sc-d0ae5099-2 cpQzcA"
      >
        <div
          class="sc-9101dc7-1 jBNtAt"
        >
          <div
            class="sc-9101dc7-0 elFxUS"
          >
            Opciones
          </div>
          <div
            class="sc-9101dc7-2"
          >
            <div
              class="sc-3a724431-0 iQabbx"
            >
              <a
                data-testid="product-list-page"
                href="/Products/List"
              >
                Products
              </a>
            </div>
          </div>
          <div
            class="sc-9101dc7-2"
          >
            <div
              class="sc-3a724431-0 iQabbx"
            >
              <a
                data-testid="cart-page"
                href="/Cart"
              >
                Cart
              </a>
            </div>
          </div>
        </div>
        <div
          class="sc-d0ae5099-3 Yvdxn"
        >
          <div>
            Content
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
